### PR TITLE
Eat the ice cream

### PR DESCRIPTION
**Pull request by Iamgoofball**
Good morning. It is time for ice cream.
:cl: Iamgoofball
add: Humans require ice cream. Eat the ice cream.
/:cl:
eattheicecream

### CODE REVIEW

**Comment by chanoc1** [comment]
tasty

**Comment by 81Denton** [comment]
that was a good video

**Comment by CitrusGender** [comment]
good meme

File: code/game/objects/items/robot/robot_items.dm [hunk 1]
```diff
@@ -448,7 +456,7 @@
 		if(R.emagged)
 			hitdamage = emaggedhitdamage
 	switch(mode)
-		if(DISPENSE_LOLLIPOP_MODE)
+		if(DISPENSE_LOLLIPOP_MODE || DISPENSE_ICECREAM_MODE)
```
Comment (SpaceManiac) [inline code/game/objects/items/robot/robot_items.dm]: `,` not `||`

**Comment by Shadowflame909** [comment]
Wtf if we buff borgies the chef wont have anything to do? I support this PR. The chef needs to have alot of ice cream. So he can make the lengthier dishes. Like cereal and pizza. 👍 Good pr goof.

**Comment by chesse20** [comment]
can we please fix hunger before we add more food

**Comment by davethwave** [comment]
Make it a upgrade module

**Comment by Kromgar** [comment]
@chesse20 shush

**Comment by yorii** [comment]
because this is is goof's first good pr.

File: code/game/objects/items/robot/robot_items.dm [hunk 2]
```diff
@@ -340,10 +340,11 @@
 #define DISPENSE_LOLLIPOP_MODE 1
 #define THROW_LOLLIPOP_MODE 2
 #define THROW_GUMBALL_MODE 3
+#define DISPENSE_ICECREAM_MODE 4
 
 /obj/item/borg/lollipop
-	name = "lollipop fabricator"
-	desc = "Reward good humans with this. Toggle in-module to switch between dispensing and high velocity ejection modes."
+	name = "treat fabricator"
+	desc = "Eat the ice cream. Toggle in-module to switch between dispensing and high velocity ejection modes."
```
Comment (Cyberboss) [inline code/game/objects/items/robot/robot_items.dm]: This should be on the icecream desc

**Review by Cyberboss — CHANGES_REQUESTED** [review]
^

File: code/game/objects/items/robot/robot_items.dm [hunk 3]
```diff
@@ -467,13 +476,17 @@
 			mode = THROW_GUMBALL_MODE
 			to_chat(user, "<span class='notice'>Module is now blasting gumballs.</span>")
 		if(THROW_GUMBALL_MODE)
+			mode = DISPENSE_ICECREAM_MODE
+			to_chat(user, "<span class='notice'>Module is now dispensing ice cream.</span>")
+		if(DISPENSE_ICECREAM_MODE)
```
Comment (Cyberboss) [inline code/game/objects/items/robot/robot_items.dm]: This will never be run

File: code/game/objects/items/robot/robot_items.dm [hunk 4]
```diff
@@ -448,7 +457,7 @@
 		if(R.emagged)
 			hitdamage = emaggedhitdamage
 	switch(mode)
-		if(DISPENSE_LOLLIPOP_MODE)
+		if(DISPENSE_LOLLIPOP_MODE, DISPENSE_ICECREAM_MODE)
```
Comment (Cyberboss) [inline code/game/objects/items/robot/robot_items.dm]: because you catch it up here

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/robot/robot_items.dm (hunk 4 above)
Comment (Iamgoofball) [reply code/game/objects/items/robot/robot_items.dm]: Different procs, dude. That's the afterattack, other one is attack_self.

**Pull request merged**